packages/realm/src/app-services/User.ts

```diff
@@ -44,11 +44,11 @@ export type UserChangeCallback = () => void;
  */
 export enum UserState {
   /** Authenticated and available to communicate with services. */
-  Active = "active",
+  LoggedIn = "LoggedIn",
   /** Logged out, but ready to be logged in. */
-  LoggedOut = "logged-out",
+  LoggedOut = "LoggedOut",
   /** Removed from the app entirely. */
-  Removed = "removed",
+  Removed = "Removed",
 }
```

Comment:
Yeah this was the most uncertain change, whether to:
* (A) Break the declared types (in order to not break the implementation of v11), or
* (B) Break the implementation (in order to fix the bug in the declared types)

Since both do break something, it's less clear which we should opt for to make a better transition from v11 - v12.

Also, if people have been using these types, they were probably depending on the implementation rather than the declarations. But I do agree about keeping the enum values lowercase, just wasn't sure if that should be a fix for v12 or v13.

What are your thoughts on (A) and (B) and how do you reason about what to break?

Reply:
Worth noting though, is the JS docs is aligned with the implementation: https://www.mongodb.com/docs/realm-sdks/js/latest/Realm.User.html#state
I'd personally vote for B as it'll make the implementation consistent with other enums and the types, which also declare our API (admittedly conflicting with both implementation and docs 😞).

Reply:
Ty for the link! Okay yeah definitely a tougher choice but interesting to get perspective on how to navigate this as well. Very helpful input 🙂 

@takameyer  and @kneth, what are your thoughts on this?

Reply:
Now that i'm looking into it, i'm very concerned about how much is mismatched regarding this state.  If one does a search in our project for [LoggedOut](https://github.com/search?q=repo%3Arealm%2Frealm-js+LoggedOut&type=code) and then a search for [logged-out](https://github.com/search?q=repo%3Arealm%2Frealm-js+logged-out&type=code) then it is visible that we are not aligned throughout the project.  

With that in mind, it is imperative that on both v11 and v12, we align with the C++ implementation (`LoggedOut`).  This could also mean a breaking change release for `realm-web` in order to stay aligned throughout all packages.  

Then I would say, we create a ticket for v13 to do a breaking change and set these to lower cased (`logged-out`).

My 2-cents

Reply:
@takameyer to ensure I understand you, which of the two alternatives suggested by LJ are you voting for? 🙂 Are you suggesting A + following up with a ticket to break this back to lower-case with v13?

> This could also mean a breaking change release for realm-web

`realm-web` is using the lower-cased variant: https://github.com/realm/realm-js/blob/main/packages/realm-web/src/User.ts#L57-L64

Reply:
A + follow up breaking change

Although we might need to break `realm-web` to make it work, if the types are being shared

Reply:
I tend to lean slightly more toward not changing the implementation as well (i.e. using `LoggedOut` etc.) for v12. But I do see both points.

To make it less annoying for users, we could actually also keep the `UserState.Active` but have the value be `LoggedIn`. This would still break the declared type, but writing `UserState.Active` would work. (Then mark it deprecated and remove it in v13):
* v12:
```ts
export enum UserState {
  /** @deprecated Will be removed in v13. Please use {@link LoggedIn} */
  Active = "LoggedIn",
  LoggedIn = "LoggedIn",
  LoggedOut = "LoggedOut",
  Removed = "Removed",
}
```
* v13:
```ts
export enum UserState {
  LoggedIn = "logged-in",
  LoggedOut = "logged-out",
  Removed = "removed",
}
```